components/tabs/src/schema.rs

```diff
@@ -45,20 +60,40 @@ impl MigrationLogic for TabsMigrationLogin {
     }
 
     fn init(&self, db: &Transaction<'_>) -> MigrationResult<()> {
-        log::debug!("Creating schema");
-        db.execute_batch(CREATE_SCHEMA_SQL)?;
+        log::debug!("Creating schemas");
+        db.execute_all(&[CREATE_SCHEMA_SQL, CREATE_META_TABLE_SQL])?;
         Ok(())
     }
 
-    fn upgrade_from(&self, _db: &Transaction<'_>, version: u32) -> MigrationResult<()> {
-        Err(MigrationError::IncompatibleVersion(version))
+    fn upgrade_from(&self, db: &Transaction<'_>, version: u32) -> MigrationResult<()> {
+        log::debug!("upgrading from {}", version);
+        match version {
+            0 => upgrade_from_v0(db),
+            _ => Err(MigrationError::IncompatibleVersion(version)),
+        }
     }
 }
 
+fn upgrade_from_v0(db: &Connection) -> MigrationResult<()> {
+    // The previous version stored the entire payload in one row
+    // and cleared on each sync -- it's fine to just drop it
+    db.execute_batch("DROP TABLE tabs;")?;
+    // Just recreate the schema
+    db.execute_batch(CREATE_SCHEMA_SQL)?;
+    Ok(())
+}
+
 #[cfg(test)]
 mod tests {
     use super::*;
     use crate::storage::TabsStorage;
+    use sql_support::open_database::test_utils::MigratedDatabaseFile;
+
+    const CREATE_V1_SCHEMA_SQL: &str = "
```

Comment:
and I guess this is v2? Maybe @bendk can help here if he remembers the semantics